lib/presenters/attachment_presenter.rb

```diff
@@ -0,0 +1,233 @@
+require "action_view"
+require "money"
+
+class AttachmentPresenter
+  attr_reader :attachment
+  include ActionView::Helpers::TagHelper
+  include ActionView::Helpers::NumberHelper
+  include ActionView::Helpers::AssetTagHelper
+
+  def initialize(attachment)
+    @attachment = attachment
+  end
+
+  def id
+    attachment.id
+  end
+
+  def order_url
+    attachment.order_url
+  end
+
+  def opendocument?
+    attachment.opendocument?
+  end
+
+  def url
+    attachment.url
+  end
+
+  def external?
+    attachment.external?
+  end
+
+  def price
+    return unless attachment.price
+    Money.from_amount(attachment.price, 'GBP').format
+  end
+
+  def accessible?
+    attachment.accessible?
+  end
+
+  def thumbnail_link
+    return if hide_thumbnail?
+    return if previewable?
+    link(attachment_thumbnail, url, "aria-hidden=true class=#{attachment_class}")
+  end
+
+  def help_block_toggle_id
+    "attachment-#{attachment.id}-accessibility-request"
+  end
+
+  def section_class
+    attachment.external? ? "hosted-externally" : "embedded"
+  end
+
+  def mail_to(email_address, name, options = {})
+    "<a href='mailto:#{email_address}?Subject=#{options[:subject]}&body=#{options[:body]}'>#{name}</a>"
+  end
+
+  def alternative_format_order_link
+    attachment_info = []
+    attachment_info << "  Title: #{attachment.title}"
+    attachment_info << "  Original format: #{attachment.file_extension}"
+    attachment_info << "  ISBN: #{attachment.isbn}" if attachment.isbn.present?
+    attachment_info << "  Unique reference: #{attachment.unique_reference}" if attachment.unique_reference.present?
+    attachment_info << "  Command paper number: #{attachment.command_paper_number}" if attachment.command_paper_number.present?
+    if attachment.hoc_paper_number.present?
+      attachment_info << "  House of Commons paper number: #{attachment.hoc_paper_number}"
+      attachment_info << "  Parliamentary session: #{attachment.parliamentary_session}"
+    end
+
+    mail_to alternative_format_contact_email, alternative_format_contact_email,
```

Comment:
yeah, parenthesis is the least of my issues with this method. 

Did a brief clean up though to *try* reduce the noise. Still this whole thing is pretty messy. We will almost certainly need to clean it up when we use it in real life. This was mostly a blind copy from whitehall job.